src/scales/scale.time.js

```diff
@@ -361,27 +361,55 @@ function generate(min, max, capacity, options) {
  * Returns the right and left offsets from edges in the form of {left, right}.
  * Offsets are added when the `offset` option is true.
  */
-function computeOffsets(table, ticks, min, max, options) {
+function computeOffsets(table, ticks, data, min, max, options) {
+	var pos = [];
+	var minInterval = 1;
+	var timeOpts = options.time;
+	var barThickness = options.barThickness;
 	var left = 0;
 	var right = 0;
-	var upper, lower;
-
-	if (options.offset && ticks.length) {
-		if (!options.time.min) {
-			upper = ticks.length > 1 ? ticks[1] : max;
-			lower = ticks[0];
-			left = (
-				interpolate(table, 'time', upper, 'pos') -
-				interpolate(table, 'time', lower, 'pos')
-			) / 2;
-		}
-		if (!options.time.max) {
-			upper = ticks[ticks.length - 1];
-			lower = ticks.length > 1 ? ticks[ticks.length - 2] : min;
-			right = (
-				interpolate(table, 'time', upper, 'pos') -
-				interpolate(table, 'time', lower, 'pos')
-			) / 2;
+	var i, ilen, curr, prev, length, width;
+
+	if (options.offset) {
+		data.forEach(function(timestamp) {
+			if (timestamp >= min && timestamp <= max) {
+				pos.push(interpolate(table, 'time', timestamp, 'pos'));
+			}
+		});
+
+		length = pos.length;
+		if (length) {
+			// Calculate minInterval
```

Comment:
This is a bit nit picky, but I would actually move this comment down a line (i.e. right above `[data, ticks].forEach(function(timestamps) {`) because right now it's not clear if it's referring to the entire block of code inside the `if (length)` statement of just to the `if (!barThickness)` section